### PR TITLE
Refactor tag methods to infer created, deleted, and kept repo objects 

### PR DESCRIPTION
**Pull request by achals**
Signed-off-by: Achal Shah <achals@gmail.com>

**What this PR does / why we need it**:

This is a first PR to support the `feast plan` command, as it applies to FCOs. It also adds a `diff_betewen` method for Registry objects, which is not complete in its implementation at this point.

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note
none
```

### CODE REVIEW

File: sdk/python/feast/diff/FcoDiff.py [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+    def __init__(self):
+        self.fco_diffs = []
+
+    def add_fco_diff(self, fco_diff: FcoDiff):
+        self.fco_diffs.append(fco_diff)
+
+
+def _tag_registry_entities_for_keep_delete(
+    existing_entities: Set[Entity], desired_entities: Set[Entity]
+) -> Tuple[Set[Entity], Set[Entity], Set[Entity]]:
```
Comment (felixwang9817) [inline sdk/python/feast/diff/FcoDiff.py:48]: in fact, can all of these methods be combined into one? e.g. `_tag_registry_objects_for_keep_delete`? seems like the internal logic is the exact same

File: sdk/python/feast/registry.py [hunk 2]
```diff
@@ -104,20 +114,60 @@ def __init__(self, registry_config: RegistryConfig, repo_path: Path):
...
+
+            self._registry_store = cls(registry_config, repo_path)
+            self.cached_registry_proto_ttl = timedelta(
+                seconds=registry_config.cache_ttl_seconds
+                if registry_config.cache_ttl_seconds is not None
+                else 0
+            )
+
+    @classmethod
+    def from_proto(cls, regsitry_proto: RegistryProto):
```
Comment (felixwang9817) [inline sdk/python/feast/registry.py:134]: nit: `regsitry` is mispelled

Comment (felixwang9817) [reply sdk/python/feast/registry.py:134]: q: why do we need a `from_proto` for the registry? I thought the registry contents would always be persisted by the registry store

to clarify, even when we want to compare two registries, we will be comparing protos; I don't think we'll need to reconstruct a registry from a proto?

**Review by felixwang9817 — CHANGES_REQUESTED** [review]

File: sdk/python/feast/diff/FcoDiff.py (hunk 1 above)
Comment (achals) [reply sdk/python/feast/diff/FcoDiff.py:48]: Yeah it's the same. I tried to do it but it made pypy very unhappy, but I can clean it up a bit and rename.

File: sdk/python/feast/registry.py (hunk 2 above)
Comment (achals) [reply sdk/python/feast/registry.py:134]: yeah fair, I can change this to operate on the protos themselves. I think I misunderstood earlier

**Review by felixwang9817 — APPROVED** [review]
/lgtm

**Pull request merged**